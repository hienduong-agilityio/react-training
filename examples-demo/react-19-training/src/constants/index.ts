export * from './api';
export * from './router';
export * from './messages';
export * from './formField';
export * from './color';
export * from './filerSearchOption';
export * from './rowPerPage';
export * from './store';
export * from './validationMessages';
