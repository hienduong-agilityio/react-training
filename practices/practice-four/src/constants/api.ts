export const HTTP_METHODS = {
  GET: 'GET',
  POST: 'POST',
  PUT: 'PUT',
  DELETE: 'DELETE'
};

export const ROOT_API = import.meta.env.VITE_API_ENDPOINT;

export const API_URL = {
  PROJECT: `${ROOT_API}/projects`
};
