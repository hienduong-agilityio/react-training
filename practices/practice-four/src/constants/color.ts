// Consider using enums instead of constants
export const COLORS_CLASS = {
  DEFAULT: 'color-default',
  PRIMARY: 'color-primary',
  SUCCESS: 'color-success',
  WARNING: 'color-warning',
  DANGER: 'color-danger '
};
