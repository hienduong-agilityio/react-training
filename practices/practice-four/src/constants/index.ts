// Index constants files
export * from './api';
export * from './color';
export * from './filerSearchOption';
export * from './route';
export * from './rowPerPage';
export * from './store';
export * from './validationMessages';
