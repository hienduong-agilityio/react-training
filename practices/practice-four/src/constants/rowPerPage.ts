export const ROWS_PER_PAGE_OPTIONS = [
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 30, label: '30' },
  { value: 40, label: '40' },
  { value: 50, label: '50' }
];
