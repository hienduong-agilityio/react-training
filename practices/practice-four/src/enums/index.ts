// Index Enum files
export * from './status';
export * from './theme';
