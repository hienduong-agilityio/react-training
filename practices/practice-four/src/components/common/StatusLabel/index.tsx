// Library
import { memo } from 'react';

// Component
import { Tag } from '@/components';

// Types
import type { ITagProps } from '@/components/common/Tag';

// Enum
import { COLORS } from '@/enums';

// Helpers
import { getColorClasses } from '@/helpers';

export interface IStatusLabelProps extends ITagProps {
  // showDot: Show dot for the StatusLabel.
  showDot?: boolean;
}

/**
 * StatusLabel component
 *
 * @returns {JSX.Element} - StatusLabel element.
 */

const StatusLabel = ({
  children,
  color = COLORS.SUCCESS,
  showDot = true,
  customClasses = ''
}: IStatusLabelProps): JSX.Element => {
  const colorClasses = getColorClasses(color);

  return (
    <div className='w-max'>
      <Tag color={color} customClasses={customClasses}>
        {showDot && (
          <svg
            role='img'
            className={`inline-flex ${colorClasses}`}
            width='6'
            height='6'
            viewBox='0 0 6 6'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
          >
            <rect width='6' height='6' rx='2' />
          </svg>
        )}
        {children}
      </Tag>
    </div>
  );
};

export default memo(StatusLabel);
