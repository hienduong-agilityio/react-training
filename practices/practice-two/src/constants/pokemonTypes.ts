import { CHIP_COLOR } from '@components/common/Chip';

const POKEMON_CHECKBOX_TYPES: {
  type: CHIP_COLOR;
  checked: boolean;
}[] = [
  { type: CHIP_COLOR.FIRE, checked: false },
  { type: CHIP_COLOR.BUG, checked: false },
  { type: CHIP_COLOR.WATER, checked: false },
  { type: CHIP_COLOR.FLYING, checked: false },
  { type: CHIP_COLOR.POISON, checked: false },
  { type: CHIP_COLOR.NORMAL, checked: false },
  { type: CHIP_COLOR.ROCK, checked: false },
  { type: CHIP_COLOR.GROUND, checked: false },
  { type: CHIP_COLOR.FIGHTING, checked: false },
  { type: CHIP_COLOR.GHOST, checked: false },
  { type: CHIP_COLOR.PSYCHIC, checked: false },
  { type: CHIP_COLOR.ICE, checked: false },
  { type: CHIP_COLOR.DRAGON, checked: false },
  { type: CHIP_COLOR.DARK, checked: false },
  { type: CHIP_COLOR.FAIRY, checked: false },
  { type: CHIP_COLOR.ELECTRIC, checked: false },
  { type: CHIP_COLOR.STEEL, checked: false },
  { type: CHIP_COLOR.GRASS, checked: false }
];

export { POKEMON_CHECKBOX_TYPES };
