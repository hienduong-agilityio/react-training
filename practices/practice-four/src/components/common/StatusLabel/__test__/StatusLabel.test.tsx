// Libraries
import { render, screen } from '@testing-library/react';

// Components
import StatusLabel, { IStatusLabelProps } from '@/components/common/StatusLabel';

// Enums
import { COLORS } from '@/enums';

// Constants
import { COLORS_CLASS } from '@/constants';

/**
StatusLabel Component
  √ should render with default props and show dot
  √ should hide the dot when showDot is false
  √ should apply custom color
  √ should apply custom classes
  √ should match snapshot
 */
describe('StatusLabel Component', () => {
  let renderComponent: (props?: Partial<IStatusLabelProps>) => ReturnType<typeof render>;

  beforeEach(() => {
    renderComponent = (props: Partial<IStatusLabelProps> = {}) => {
      return render(<StatusLabel {...props}>Status label</StatusLabel>);
    };
  });

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render with default props and show dot', () => {
    renderComponent();

    const dotElement = screen.getByRole('img');

    expect(dotElement).toBeInTheDocument();
    expect(dotElement).toHaveClass(`inline-flex color-success`);
  });

  it('should hide the dot when showDot is false', () => {
    renderComponent({ showDot: false });

    const dotElement = screen.queryByRole('img');

    expect(dotElement).not.toBeInTheDocument();
  });

  it('should apply custom color', () => {
    renderComponent({ color: COLORS.DANGER });

    const dotElement = screen.getByRole('img');

    expect(dotElement).toHaveClass(COLORS_CLASS.DANGER);
  });

  it('should apply custom classes', () => {
    const customClass = 'my-custom-class';
    renderComponent({ customClasses: customClass });

    const tagElement = screen.getByText('Status label').closest('span');

    expect(tagElement).toHaveClass(customClass);
  });
});
