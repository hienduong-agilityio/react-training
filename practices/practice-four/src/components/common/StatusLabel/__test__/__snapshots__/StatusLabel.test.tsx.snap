// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel Component should match snapshot 1`] = `
<div>
  <div
    class="w-max"
  >
    <span
      class="inline-flex gap-[6px] rounded font-medium leading-[18px] tracking-[0.36px] flex items-center justify-center color-success"
    >
      <svg
        class="inline-flex color-success"
        fill="none"
        height="6"
        role="img"
        viewBox="0 0 6 6"
        width="6"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="6"
          rx="2"
          width="6"
        />
      </svg>
      Status label
    </span>
  </div>
</div>
`;
